core-web/libs/sdk/experiments/src/lib/dot-experiments.ts

```diff
@@ -0,0 +1,155 @@
+import { API_EXPERIMENTS_URL, EXPERIMENT_DB_KEY_PATH, EXPERIMENT_DB_STORE_NAME } from './constants';
+import { AssignedExperiments, IsUserIncludedApiResponse, SdkExperimentConfig } from './models';
+import { IndexDBDatabaseHandler } from './persistence/index-db-database-handler';
+import { dotLogger } from './utils/utils';
+
+/**
+ * `SdkExperiments` is a Typescript class to handles all operations related to fetching, storing, parsing, and navigating
+ * data for Experiments (A/B Testing).
+ *
+ * It requires a configuration object for instantiation, please instance it using the method `getInstance` sending
+ * an object with `api-key`, `server` and `debug`.
+ *
+ * Here's an example of how you can instantiate SdkExperiments class:
+ * @example
+ * ```typescript
+ * const instance = SdkExperiments.getInstance({
+ *   server: "yourServerUrl",
+ *   "api-key": "yourApiKey"
+ * });
+ * ```
+ *
+ * @export
+ * @class DotExperiments
+ *
+ */
+export class DotExperiments {
+    // TODO: make this class the entrypoint of the library
+    private static instance: DotExperiments;
+
+    private constructor(private config: SdkExperimentConfig) {
+        if (!this.config['server']) {
+            throw new Error('`server` must be provided and should not be empty!');
+        }
+
+        if (!this.config['api-key']) {
```

Comment:
Maybe we can create constants or an enum for this index accessors `server/api-key`